[datadog_tag_pipeline_ruleset] Enforce canonical query form at plan time for tag pipeline resources (#3912)
DataDog/terraform-provider-datadog: docs/resources/tag_pipeline_ruleset.md

- `case_insensitivity` (Boolean) Whether the query matching is case insensitive.
- `if_not_exists` (Boolean, Deprecated) Whether to apply the query only if the key doesn't exist.
- `if_tag_exists` (String) Behavior when the tag already exists. Valid values: `append` (append to the existing tag value), `replace` (replace existing tag value), `do_not_apply` (never apply if tag already exists). Valid values are `append`, `replace`, `do_not_apply`.

> - `query` (String) The query string. Must be in canonical form. Datadog normalizes queries (operator casing, spacing, redundant parentheses, quoting, and similar), and the provider validates the configured value against this canonical form during planning. If they differ, the plan fails with an error showing the canonical query to use in the configuration.

Reviewer: ```suggestion
- `query` (String) The query string. Datadog normalizes queries to a canonical form (operator casing, spacing, redundant parentheses, quoting, and similar formatting differences). The provider validates the configured value against this canonical form during planning. If the value doesn't match, the plan fails with an error showing the canonical query to use in the configuration.
```
